hazelcast/src/hazelcast/client/cluster/impl/VectorClock.cpp

```diff
@@ -27,41 +25,45 @@ namespace hazelcast {
 
                 VectorClock::VectorClock() {}
 
-                typedef std::vector<std::pair<std::string, boost::shared_ptr<int64_t> > > TimestampVector;
+                VectorClock::VectorClock(const VectorClock::TimestampVector &replicaLogicalTimestamps) {
+                    BOOST_FOREACH(const VectorClock::TimestampVector::value_type &replicaTimestamp,
+                                  replicaLogicalTimestamps) {
+                                    replicaTimestamps[replicaTimestamp.first] = replicaTimestamp.second;
+                                }
+                }
 
-                std::vector<std::pair<std::string, int64_t> > VectorClock::entrySet() {
-                    TimestampVector entries = replicaTimestamps.entrySet();
-                    std::vector<std::pair<std::string, int64_t> > result;
-                    BOOST_FOREACH(const TimestampVector::value_type &entry , entries) {
-                        result.push_back(std::make_pair(entry.first, *entry.second));
+                VectorClock::TimestampVector VectorClock::entrySet() {
+                    VectorClock::TimestampVector result;
```

Comment:
Please consider reusing TimestampVector passed as a constructor argument. This way you avoid recreating it from map/set

Reply:
Changed as suggested.